widget/embedded/src/components/CustomTokenModal/CustomTokenModal.tsx

```diff
@@ -0,0 +1,100 @@
+import type { PropTypes } from './CustomTokenModal.types';
+
+import { i18n } from '@lingui/core';
+import {
+  Button,
+  Divider,
+  ExternalLinkIcon,
+  Image,
+  Typography,
+} from '@rango-dev/ui';
+import React from 'react';
+
+import { getContainer } from '../../utils/common';
+import { WatermarkedModal } from '../common/WatermarkedModal';
+
+import { Container, StyledLink } from './CustomTokenModal.styles';
+
+export function CustomTokenModal(props: PropTypes) {
+  const { open, onClose, token, handleSubmitClick, blockchain } = props;
+
+  const shouldShowAddressLink =
```

Comment:
Please replace `shouldShow...` and `getFormatted...` with following changes.

Create a `CustomTokenModal.helpers.ts` then:
```

export function generateExplorerLink(
  address: string,
  blockchain: BlockchainMeta
): string {
  // We don't have Cosmos explorer url in API.
  if (blockchain.type === 'COSMOS') {
    return '';
  }

  const explorerLinkPattern = blockchain.info?.addressUrl;

  if (!explorerLinkPattern) {
    return '';
  }

  return explorerLinkPattern.replace('{wallet}', address);
};
```

Then, in this component:


```
export function CustomTokenModal(props: PropTypes) {
  // ...
   const explorerLink = generateExplorerLink(token.address);
   // ..
}
```

Reply:
fixed